tsl/src/compression/compression_dml.c

```diff
@@ -599,6 +649,16 @@ batch_matches(RowDecompressor *decompressor, ScanKeyData *scankeys, int num_scan
 #endif
 		if (valid)
 		{
+			if (on_conflict == ONCONFLICT_NONE)
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_UNIQUE_VIOLATION),
+						 errmsg("duplicate key value violates unique constraint")));
+			}
+			if (on_conflict == ONCONFLICT_NOTHING)
+			{
+				*skip_insert = true;
+			}
```

Comment:
Should we do it in the caller instead, based on the return value? This interface looks a little too complicated -- the function has two bool return values and can also throw an error.

Reply:
I initiallly had this in caller but then i would have two implementations 1 for index and 1 for seq scan so i moved it

Reply:
I would copy these lines instead of having a complicated interface, but anyway, not a big deal.